util/io/src/service_mio.rs

```diff
@@ -267,7 +267,8 @@ impl<Message> Handler for IoManager<Message> where Message: Send + Clone + Sync
 				event_loop.shutdown();
 			},
 			IoMessage::AddHandler { handler } => {
-				let handler_id = self.handlers.write().insert(handler.clone()).unwrap_or_else(|_| panic!("Too many handlers registered"));
+				let handler_id = self.handlers.write().insert(handler.clone());
+				assert!(handler_id <= MAX_HANDLERS, "Too many handlers registered");
```

Comment:
Old handler ids are re-used?

Reply:
Yes, when we remove a handler its ID becomes available again.
In practice it doesn't really matter since we just add all the handlers at once at initialization, and remove them all at once at the end.